### PR TITLE
feature : added file sharing feature.

### LINKED ISSUE #450: Develop the Share feature in the app
**Actual Behaviour.**  
Missing Share feature in the app.

**Expected Behaviour**  
Develop the Share feature in the app.

**Steps to reproduce it**  
Check the bottom option in the navigation drawer of the app.

**Additional context**  
By share, we mean sharing a logged/recorded/test dataset to another user's device.

**Would you like to work on the issue?**
Issue for Codeheat.

### PR DESCRIPTION
**Pull request by dipu989**
Fixes #450 

**Changes**: 
Added Feature for file sharing.

**Gif for the changes**:
sharegif

**Checklist**: 
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**:
[apkDebug.zip](https://github.com/fossasia/neurolab-android/files/3718370/apkDebug.zip)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -65,6 +67,7 @@ dependencies {
     implementation "com.opencsv:opencsv:$rootProject.opencsvLibraryVersion", {
         exclude group: "commons-logging", module: "commons-logging"
     }
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Comment (jddeep) [inline app/build.gradle]: I believe it is already present above. Please check.

**Review by jddeep — CHANGES_REQUESTED** [review]

**Comment by dipu989** [comment]
@jddeep  Done with the changes! Please take a look.

**Comment by jddeep** [comment]
@dipu989 Please pull latest changes and rerun the build here. Thanks.

**Comment by dipu989** [comment]
@jddeep Done with it. Checks passed!

**Comment by adityastic** [comment]
@dipu989 semantic pr title :)

**Comment by dipu989** [comment]
> @dipu989 semantic pr title :)
Will keep this in mind for sure from the next time!

**Review by jddeep — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -23,7 +23,19 @@
...
+
+        <provider
+            android:name="android.support.v4.content.FileProvider"
+            android:authorities="io.neurolab.fileprovider"
+            android:exported="false"
+            android:grantUriPermissions="true">
+            <meta-data
+                android:name="android.support.FILE_PROVIDER_PATHS"
+                android:resource="@xml/filepaths" />
+        </provider>
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml]: Keep activity tags together and move this block to a place after them.

File: app/src/main/java/io/neurolab/activities/ShareDataActivity.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+               }
+           }
+          updateSelectedFiles(selectedFiles);
+          setPermission();
+         }
+       }
+      );}
+
+      private void setPermission()
+      {
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/activities/ShareDataActivity.java]: Reformat code

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -50,6 +50,8 @@
     <string name="focus_desc">Test the focus strength of your brain. Improve your score with more new sessions.</string>
     <string name="relax_desc">Relax yourself. Regular sessions can help level up your concentration.</string>
 
+    <string name="share_toast">Please select at least one record!</string>
+    <string name="share_screen_toast">Tap to select and Share!</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Remove the `!`s

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (dipu989) [reply app/src/main/res/values/strings.xml]: Ok! On it. :+1:

**Comment by dipu989** [comment]
@CloudyPadmal Changed as per review. Please check!

File: app/src/main/java/io/neurolab/activities/ShareDataActivity.java (hunk 3 above)
Comment (dipu989) [reply app/src/main/java/io/neurolab/activities/ShareDataActivity.java]: @CloudyPadmal I have changed according to review.
Can you please check?

**Review by CloudyPadmal — APPROVED** [review]

**Pull request merged**